azext_edge/edge/providers/rpsaas/adr/assets.py

```diff
@@ -368,16 +368,20 @@ def add_dataset_data_point(
         observability_mode: Optional[str] = None,
         queue_size: Optional[int] = None,
         sampling_interval: Optional[int] = None,
+        replace: bool = False
     ):
         asset = self.show(
             asset_name=asset_name,
             resource_group_name=resource_group_name,
             check_cluster=True
         )
         dataset = _get_dataset(asset, dataset_name, create_if_none=True)
-        if not dataset.get("dataPoints"):
-            dataset["dataPoints"] = []
-
+        dataset["dataPoints"] = dataset.get("dataPoints", [])
+        point_names = [point["name"] for point in dataset["dataPoints"]]
+        if not replace and data_point_name in point_names:
+            raise InvalidArgumentValueError(
+                DUPLICATE_POINT_ERROR.format(data_point_name)
```

Comment:
I thought the behavior was to ignore duplicates if replace was not provided not raise an error.